Std/Data/HashMap/Basic.lean

```diff
@@ -291,51 +317,129 @@ replaced with `a => b`.
 
 /--
 Removes key `a` from the map. If it does not exist in the map, the map is returned unchanged.
+```
+def hashMap := ofList [("one", 1), ("two", 2)]
+hashMap.erase "one" = ofList [("two", 2)]
+hashMap.erase "three" = ofList [("one", 1), ("two", 2)]
+```
 -/
 @[inline] def erase (self : HashMap α β) (a : α) : HashMap α β := ⟨self.1.erase a, self.2.erase⟩
 
 /--
 Performs an in-place edit of the value, ensuring that the value is used linearly.
 The function `f` is passed the original key of the entry, along with the value in the map.
+```
+(ofList [("one", 1), ("two", 2)]).modify "one" (fun _ v => v + 1) = ofList [("one", 2), ("two", 2)]
```

Comment:
Given that there is no instance of `BEQ` for `HashMap` this an acceptable way to write this example? Would it be better to use `=>` instead of `=`?

Reply:
(maybe we'll make that actual syntax at some point, it would be nice to have a way to create concrete hashmaps that way)

I agree that it would be best if the LHS at least was actual code, because even if it's pseudocode now we still want to test it (occasionally these doctests reveal issues in either the doc or the function itself).

Reply:
`↦` was causing the [Check for forbidden character ↦](https://github.com/leanprover/std4/blob/main/.github/workflows/build.yml#L41) CI step to fail.

Replaced `↦` with `=>` to align with #252